pkg/nodeconfig/nodeconfig.go

```diff
@@ -664,13 +664,34 @@ func (nc *nodeConfig) UpdateKubeletClientCA(contents []byte) error {
 	return nil
 }
 
+// bootstrap starts all required services to bootstrap a node object
+func (nc *nodeConfig) bootstrap() error {
+	if err := nc.Windows.Prepare(); err != nil {
+		return err
+	}
+	saCA, saToken, err := getWICDCertAndToken(nc.k8sclientset)
+	if err != nil {
+		return err
+	}
+	return nc.Windows.Bootstrap(version.Get(), nodeConfigCache.apiServerEndpoint, saCA, saToken)
+}
+
 // configureWICD configures and ensures WICD is running
 func (nc *nodeConfig) configureWICD() error {
+	saCA, saToken, err := getWICDCertAndToken(nc.k8sclientset)
+	if err != nil {
+		return err
+	}
+	return nc.Windows.ConfigureWICD(nodeConfigCache.apiServerEndpoint, saCA, saToken)
+}
+
+// getWICDTokenAndCert returns the token and CA cert associated with the WICD service account
+func getWICDCertAndToken(clientset *kubernetes.Clientset) ([]byte, []byte, error) {
```

Comment:
This can be made a method of nodeConfig.

Can the cached client be used for this now that you've added that to the nodeConfig struct?